docs/_ext/autoclass_with_examples.py

```diff
@@ -100,14 +94,18 @@ def _create_hint(file_name: str, method_name: str) -> str:
         """
 
 
-def _append_hint(method_name: str, method, base_class: Any, hint: str):
+def _append_hint(method: Any, class_: Any, hint: str) -> None:
     """
-    If method does not have the __doc__, takes it from the base method.
+    If method does not have the __doc__, takes it from the ancestor method.
     """
-    if method.__doc__ is None and method_name in base_class.__dict__:
-        method.__doc__ = getattr(base_class, method_name).__doc__ + hint
-    else:
-        method.__doc__ = (method.__doc__ or "") + hint
+    parent_method = method
+    while parent_method is not None and parent_method.__doc__ is None:
+        class_ = class_.__base__
+        parent_method = getattr(class_, method.__name__, None)
+
+    if parent_method is None:
+        method.__doc__ = hint
+    method.__doc__ = (parent_method.__doc__ or "") + hint
```

Comment:
If `parent_method` is None it will fail. Probably a return under the `method.__doc__ = hint` will fix the issue

Reply:
Actually, `None.__doc__` returns `None`. But I'll reorder the if statement.